### ISSUE TITLE
[Monitoring] Watches not guarding against all null values

### ISSUE THREAD

**Issue by cachedout** [elastic/elasticsearch#48981]
Using the default monitoring watches, users report that occasionally an error similar to the following is logged:

```
[es/i-1/es.log] [2019-10-15T00:00:20,089][ERROR][org.elasticsearch.xpack.watcher.transform.script.ExecutableScriptTransform] [instance-0000000001] failed to execute [script] transform for [YUbZjlbDSl-JSV1-XZpICQ_elasticsearch_cluster_status_eabb3c24-1fca-46e8-92ea-04ffe926e15c-2019-10-15T00:00:19.596289Z]
org.elasticsearch.script.ScriptException: runtime error
        at org.elasticsearch.painless.PainlessScript.convertToScriptException(PainlessScript.java:94) ~[?:?]
        at org.elasticsearch.painless.PainlessScript$Script.execute(ctx.vars.email_recipient = (ctx.payload.kibana_settings.hits.total > 0 && ctx.payload.kibana_settings.hits.hits[0]._source.kibana_settings.xpack != null) ? ctx.payload.kibana_settings.hits.hits[0]._source.kibana_settings.xpack.default_admin_email : null;ct ...:1152) ~[?:?]
        at org.elasticsearch.xpack.watcher.transform.script.ExecutableScriptTransform.doExecute(ExecutableScriptTransform.java:49) ~[x-pack-watcher-7.2.0.jar:7.2.0]
        at org.elasticsearch.xpack.watcher.transform.script.ExecutableScriptTransform.execute(ExecutableScriptTransform.java:38) [x-pack-watcher-7.2.0.jar:7.2.0]
        at org.elasticsearch.xpack.watcher.transform.script.ExecutableScriptTransform.execute(ExecutableScriptTransform.java:23) [x-pack-watcher-7.2.0.jar:7.2.0]
        at org.elasticsearch.xpack.watcher.execution.ExecutionService.executeInner(ExecutionService.java:505) [x-pack-watcher-7.2.0.jar:7.2.0]
        at org.elasticsearch.xpack.watcher.execution.ExecutionService.execute(ExecutionService.java:309) [x-pack-watcher-7.2.0.jar:7.2.0]
        at org.elasticsearch.xpack.watcher.execution.ExecutionService.lambda$executeAsync$5(ExecutionService.java:410) [x-pack-watcher-7.2.0.jar:7.2.0]
        at org.elasticsearch.xpack.watcher.execution.ExecutionService$WatchExecutionTask.run(ExecutionService.java:605) [x-pack-watcher-7.2.0.jar:7.2.0]
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:688) [elasticsearch-7.2.0.jar:7.2.0]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
        at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64) ~[?:?]
        at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70) ~[?:?]
        at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248) ~[?:?]
        at java.util.Objects.checkIndex(Objects.java:372) ~[?:?]
        at java.util.ArrayList.get(ArrayList.java:458) ~[?:?]
        at org.elasticsearch.painless.PainlessScript$Script.execute(ctx.vars.email_recipient = (ctx.payload.kibana_settings.hits.total > 0 && ctx.payload.kibana_settings.hits.hits[0]._source.kibana_settings.xpack != null) ? ctx.payload.kibana_settings.hits.hits[0]._source.kibana_settings.xpack.default_admin_email : null;ct ...:429) ~[?:?]
        ... 11 more
```

It appears as though there is a bug in the painless script somewhere wherein we are not properly checking for empty lists. Investigation has not yet surfaced an exact location in the script for this problem, however. This issue is being filed to track further investigation of this bug and to add better safety checks to the script for empty values.

**Comment by jakelandis** [comment]
@cachedout - I think this is a duplicate of what https://github.com/elastic/elasticsearch/pull/45308 fixes. However, I would not be too surprised if there are additional places where an IndexOutOfBounds could occur.  

Since the line numbers and stack trace look identical, I am going to close this a resolved as of 7.5.0 with that PR. If you think otherwise, please re-open.

**Comment by cachedout** [comment]
@jakelandis Ah, very good! Thanks for the heads up. Much appreciated.
